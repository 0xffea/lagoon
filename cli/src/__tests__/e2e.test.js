// @flow

import path from 'path';
import { sync as spawnSync } from 'execa';

const CLI_PATH = path.join(__dirname, '..', '..', 'bin', 'lagu.js');

describe('lagu', () => {
  it('should fail with error message without any arguments', async () => {
    const results = spawnSync(CLI_PATH, [], { reject: false });
    expect(results.code).toBe(1);
    expect(results.message).toMatch('Not enough non-option arguments');
  });

  it('should show customer details (using --project option)', async () => {
    const results = spawnSync(
      CLI_PATH,
      ['customer', '--project', 'ci-multiproject1'],
      {
        reject: false,
      },
    );
    expect(results.code).toBe(0);
    expect(results.stdout).toMatchSnapshot();
  });

  // TODO: Run this in a directory that has a fixture .lagoon.yml file
  it('should show customer details (project read from .lagoon.yml)', async () => {
    const results = spawnSync(CLI_PATH, ['customer'], {
      reject: false,
    });
    expect(results.code).toBe(0);
    expect(results.stdout).toMatchSnapshot();
  });

  // TODO: Comment in, figure out why this returns nothing (do we need GraphQL fixture data?)
  // it('should list all environments for a project (using --project option)', async () => {
  //   const results = spawnSync(CLI_PATH, ['environments', '--project', 'ci-multiproject1'], {
  //     reject: false,
  //   });
  //   expect(results.code).toBe(0);
  //   expect(results.stdout).toMatchSnapshot();
  // });
  // // TODO: Run this in a directory that has a fixture .lagoon.yml file
  // it('should list all environments for a project (project read from .lagoon.yml)', async () => {
  //   const results = spawnSync(CLI_PATH, ['environments'], {
  //     reject: false,
  //   });
  //   expect(results.code).toBe(0);
  //   expect(results.stdout).toMatchSnapshot();
  // });

  it('should show project details (using --project option)', async () => {
    const results = spawnSync(
      CLI_PATH,
      ['project', '--project', 'ci-multiproject1'],
      {
        reject: false,
      },
    );
    expect(results.code).toBe(0);
    expect(results.stdout).toMatchSnapshot();
  });

  // TODO: Run this in a directory that has a fixture .lagoon.yml file
  it('should show project details (read from .lagoon.yml)', async () => {
    const results = spawnSync(CLI_PATH, ['project'], {
      reject: false,
    });
    expect(results.code).toBe(0);
    expect(results.stdout).toMatchSnapshot();
  });

  it('should list all projects', async () => {
    const results = spawnSync(CLI_PATH, ['projects'], { reject: false });
    expect(results.code).toBe(0);
    expect(results.stdout).toMatchSnapshot();
  });
});
