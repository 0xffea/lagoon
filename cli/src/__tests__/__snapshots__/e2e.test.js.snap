// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli should list all projects 1`] = `
"╔════════════════════╤═════════════════════════════════════════════════════╤══════════╤═══════════════╤═════════════════════╗
║ Project            │ Git URL                                             │ Branches │ Pull Requests │ Created             ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-bitbucket       │ ssh://git@192.168.99.1:2222/git/bitbucket.git       │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-drupal          │ ssh://git@192.168.99.1:2222/git/drupal.git          │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-drupal-galera   │ ssh://git@192.168.99.1:2222/git/drupal-galera.git   │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-drupal-postgres │ ssh://git@192.168.99.1:2222/git/drupal-postgres.git │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-elasticsearch   │ ssh://git@192.168.99.1:2222/git/elasticsearch.git   │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-features        │ ssh://git@192.168.99.1:2222/git/features.git        │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-github          │ ssh://git@192.168.99.1:2222/git/github.git          │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-gitlab          │ ssh://git@192.168.99.1:2222/git/gitlab.git          │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-multiproject1   │ ssh://git@192.168.99.1:2222/git/multiproject.git    │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-multiproject2   │ ssh://git@192.168.99.1:2222/git/multiproject.git    │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-nginx           │ ssh://git@192.168.99.1:2222/git/nginx.git           │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-node            │ ssh://git@192.168.99.1:2222/git/node.git            │ true     │ true          │ 2018-06-08 15:20:52 ║
╟────────────────────┼─────────────────────────────────────────────────────┼──────────┼───────────────┼─────────────────────╢
║ ci-rest            │ ssh://git@192.168.99.1:2222/git/rest.git            │ true     │ true          │ 2018-06-08 15:20:52 ║
╚════════════════════╧═════════════════════════════════════════════════════╧══════════╧═══════════════╧═════════════════════╝
"
`;

exports[`cli should show customer details (project read from .lagoon.yml) 1`] = `
"Customer details for project 'ci-multiproject1':
╔════════════════════╤═════════════════════╗
║ Name               │ ci-customer         ║
╟────────────────────┼─────────────────────╢
║ Comment            │                     ║
╟────────────────────┼─────────────────────╢
║ Deploy Private Key │ √                   ║
╟────────────────────┼─────────────────────╢
║ SSH Keys           │ ci-customer-sshkey  ║
╟────────────────────┼─────────────────────╢
║ Created            │ 2018-06-08 15:20:52 ║
╚════════════════════╧═════════════════════╝
"
`;

exports[`cli should show customer details (using --project option) 1`] = `
"Customer details for project 'ci-multiproject1':
╔════════════════════╤═════════════════════╗
║ Name               │ ci-customer         ║
╟────────────────────┼─────────────────────╢
║ Comment            │                     ║
╟────────────────────┼─────────────────────╢
║ Deploy Private Key │ √                   ║
╟────────────────────┼─────────────────────╢
║ SSH Keys           │ ci-customer-sshkey  ║
╟────────────────────┼─────────────────────╢
║ Created            │ 2018-06-08 15:20:52 ║
╚════════════════════╧═════════════════════╝
"
`;

exports[`cli should show project details (read from .lagoon.yml) 1`] = `
"Project details for 'ci-multiproject1':
╔═══════════════════════╤══════════════════════════════════════════════════╗
║ Project Name          │ ci-multiproject1                                 ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Customer              │ ci-customer                                      ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Git URL               │ ssh://git@192.168.99.1:2222/git/multiproject.git ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Active Systems Deploy │ lagoon_openshiftBuildDeploy                      ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Active Systems Remove │ lagoon_openshiftRemove                           ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Branches              │ true                                             ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Pull Requests         │ true                                             ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Openshift             │ ci-local                                         ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Created               │ 2018-06-08 15:20:52                              ║
╚═══════════════════════╧══════════════════════════════════════════════════╝
"
`;

exports[`cli should show project details (using --project option) 1`] = `
"Project details for 'ci-multiproject1':
╔═══════════════════════╤══════════════════════════════════════════════════╗
║ Project Name          │ ci-multiproject1                                 ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Customer              │ ci-customer                                      ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Git URL               │ ssh://git@192.168.99.1:2222/git/multiproject.git ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Active Systems Deploy │ lagoon_openshiftBuildDeploy                      ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Active Systems Remove │ lagoon_openshiftRemove                           ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Branches              │ true                                             ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Pull Requests         │ true                                             ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Openshift             │ ci-local                                         ║
╟───────────────────────┼──────────────────────────────────────────────────╢
║ Created               │ 2018-06-08 15:20:52                              ║
╚═══════════════════════╧══════════════════════════════════════════════════╝
"
`;
